Automattic/wp-calypso
# Launchpad: Update the navigator data store

Related to https://github.com/Automattic/wp-calypso/pull/81576

## Proposed Changes

* This is a follow-up PR from the notes Dale posted on https://github.com/Automattic/wp-calypso/pull/81576#pullrequestreview-1631682563. I made a few changes to the data structure to improve the extensibility of the code.
* Also fixes the type for the `available_checklists` data return from the `GET /sites/${ slug }/launchpad/navigator`.

## Testing Instructions



* For now, we only have the use case for setting the active task list
* To test it, apply this branch or use the calypso live link
* Make sure you are sandboxed and have the https://github.com/Automattic/jetpack/pull/32964 changes applied
* Create a new site or use an existing one that has a post-launch Launchpad
* Click on a task. It should make a `PUT` request to `/launchpad/navigator`

## Pre-merge Checklist
